packages/flutter_tools/test/general.shard/build_system/targets/dart_plugin_registrant_test.dart

```diff
@@ -0,0 +1,226 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'dart:io';
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_system/build_system.dart';
+import 'package:flutter_tools/src/build_system/targets/dart_plugin_registrant.dart';
+import 'package:flutter_tools/src/project.dart';
+
+import '../../../src/common.dart';
+import '../../../src/context.dart';
+
+const String _kSamplePackageJson = '''
+{
+  "configVersion": 2,
+  "packages": [
+    {
+      "name": "path_provider_linux",
+      "rootUri": "/path_provider_linux",
+      "packageUri": "lib/",
+      "languageVersion": "2.12"
+    }
+  ]
+}
+''';
+
+const String _kSamplePackagesFile = '''
+path_provider_linux:/path_provider_linux/lib/
+''';
+
+const String _kSamplePackageJsonWindows = '''
+{
+  "configVersion": 2,
+  "packages": [
+    {
+      "name": "path_provider_linux",
+      "rootUri": "file:///C:/path_provider_linux",
+      "packageUri": "lib/",
+      "languageVersion": "2.12"
+    }
+  ]
+}
+''';
+
+const String _kSamplePackagesFileWindows = '''
+path_provider_linux:file:///C:/path_provider_linux/lib/
+''';
+
+const String _kSamplePubspecFile = '''
+name: path_provider_example
+description: Demonstrates how to use the path_provider plugin.
+
+dependencies:
+  flutter:
+    sdk: flutter
+  path_provider_linux: 1.0.0
+''';
+
+const String _kLinuxRegistrant =
+'//\n'
+'// Generated file. Do not edit.\n'
+'//\n'
+'\n'
+'// @dart = 2.12\n'
+'\n'
+'import \'\' as entrypoint;\n'
+'import \'dart:io\'; // flutter_ignore: dart_io_import.\n'
+'import \'package:path_provider_linux/path_provider_linux.dart\';\n'
+'\n'
+'@pragma(\'vm:entry-point\')\n'
+'class _PluginRegistrant {\n'
+'\n'
+'  @pragma(\'vm:entry-point\')\n'
+'  static void register() {\n'
+'    if (Platform.isLinux) {\n'
+'      try {\n'
+'        PathProviderLinux.registerWith();\n'
+'      } catch (err) {\n'
+'        print(\n'
+'          \'`path_provider_linux` threw an error: \$err. \'\n'
+'          \'The app may not function as expected until you remove this plugin from pubspec.yaml\'\n'
+'        );\n'
+'      }\n'
+'\n'
+'    } else if (Platform.isMacOS) {\n'
+'    } else if (Platform.isWindows) {\n'
+'    }\n'
+'  }\n'
+'\n'
+'}\n'
+'\n'
+'void main() {\n'
+'  entrypoint.main();\n'
+'}\n'
+'';
+
+const String _kSamplePluginPubspec = '''
+name: path_provider_linux
+description: linux implementation of the path_provider plugin
+// version: 2.0.1
+// homepage: https://github.com/flutter/plugins/tree/master/packages/path_provider/path_provider_linux
+
+flutter:
+  plugin:
+    implements: path_provider
+    platforms:
+      linux:
+        dartPluginClass: PathProviderLinux
+        pluginClass: none
+
+environment:
+  sdk: ">=2.12.0-259.9.beta <3.0.0"
+  flutter: ">=1.20.0"
+''';
+
+void main() {
```

Comment:
consider the following scenario:

* User has pubspec dependencies that require registrant generation and builds
* User modifies pubspec to no longer have these dependencies

Can you write a test that verifies that the generated registrant gets deleted correctly?

Reply:
Test was added. Done.